Issue/5599 Modal and popover presentation updates

frosty opened the pull request:
Refs #5599. 

This picks off some of the low-hanging fruit listed in the above issue around some more appropriate uses of modals and popovers on iPad.

### To test

Check the following:

* People and Posts filter dropdowns from the navigation bar are popovers on all devices (and no longer include a navigation bar – fixes #3748).
* Adding a category (either from Post Settings > Categories or from Site Settings > Default Category) should now present a form sheet modal on iPad (full screen on phone)
* Adding a featured image and a location to a post (in post settings) should now use form sheet modals on iPad.

Needs review: @kurzee

kurzee commented:
One issue, when trying to search locations on the iPad form sheet:

![simulator screen shot jun 29 2016 1 32 57 pm](https://cloud.githubusercontent.com/assets/1873422/16464067/27225f1a-3dfe-11e6-81f7-c1b39c5d09b8.png)

Reproduce by tapping and activating the text field.

Note: does not reproduce on the previous full screen modal.

kurzee commented on WordPress/Classes/ViewRelated/System/ForcePopoverPresentationPopoverControllerDelegate.swift:
```diff
@@ -0,0 +1,30 @@
+import UIKit
+
+/// This delegate forces popover presentation even on iPhone / in compact size classes
+class ForcePopoverPresentationPopoverControllerDelegate: NSObject, UIPopoverPresentationControllerDelegate {
```
This class name is a bit intense 😄, just spitballing here, but what about `ForcedPopoverPresentation` since it kind of acts as more of a helper than an object. 🤔

frosty replied:
Haha, `ForcedPopoverPresenter` maybe?

kurzee replied:
I like the sound of that.

Perhaps we might also separate the two, with `ForcedPopoverPresenter` and separately an object that conforms to `UIPopoverPresentationControllerDelegate`. Maybe this way the `ForcedPopoverPresenter` would maintain a conforming object of something like `ForcedPopoverPresenterDelegate`, separating the presenter and delegate handler. I dunno, that may not be very swifty.

kurzee commented:
@frosty other than the search issue, everything else looks good 🎉.

Let me know when ready for a second look on the search bug.

frosty commented:
@kurzee Ping, back to you! I fixed the search bar issue (if you could double-check this on both iPad and iPhone that'd be good), and renamed the ForcePopover... thing.

kurzee commented:
@frosty awesome, good to go! :shipit:
